File: MyStore/MyStore.Client/Content/fsharp/MyStore.Domain/Dto/Shop.js
import { Record } from "../../.fable/fable-library.3.1.11/Types.js";
import { array_type, bool_type, class_type, record_type, float64_type, string_type, int32_type } from "../../.fable/fable-library.3.1.11/Reflection.js";

export class ProductDto extends Record {
    constructor(productId, name, description, price) {
        super();
        this.productId = (productId | 0);
        this.name = name;
        this.description = description;
        this.price = price;
    }
}

export function ProductDto$reflection() {
    return record_type("MyStore.Dto.Shop.ProductDto", [], ProductDto, () => [["productId", int32_type], ["name", string_type], ["description", string_type], ["price", float64_type]]);
}

export class CustomerDto extends Record {
    constructor(customerId, firstName, lastName, honorific, email, currentCartId) {
        super();
        this.customerId = (customerId | 0);
        this.firstName = firstName;
        this.lastName = lastName;
        this.honorific = honorific;
        this.email = email;
        this.currentCartId = currentCartId;
    }
}

export function CustomerDto$reflection() {
    return record_type("MyStore.Dto.Shop.CustomerDto", [], CustomerDto, () => [["customerId", int32_type], ["firstName", string_type], ["lastName", string_type], ["honorific", string_type], ["email", string_type], ["currentCartId", class_type("System.Nullable`1", [int32_type])]]);
}

export class CartDto extends Record {
    constructor(cartId, isPublic, ownerCustomerId) {
        super();
        this.cartId = (cartId | 0);
        this.isPublic = isPublic;
        this.ownerCustomerId = ownerCustomerId;
    }
}

export function CartDto$reflection() {
    return record_type("MyStore.Dto.Shop.CartDto", [], CartDto, () => [["cartId", int32_type], ["isPublic", bool_type], ["ownerCustomerId", class_type("System.Nullable`1", [int32_type])]]);
}

export class OrderDto extends Record {
    constructor(orderId, customerId, createTimeOffset) {
        super();
        this.orderId = (orderId | 0);
        this.customerId = (customerId | 0);
        this.createTimeOffset = createTimeOffset;
    }
}

export function OrderDto$reflection() {
    return record_type("MyStore.Dto.Shop.OrderDto", [], OrderDto, () => [["orderId", int32_type], ["customerId", int32_type], ["createTimeOffset", class_type("System.DateTimeOffset")]]);
}

export class OrderedProductDto extends Record {
    constructor(productId, orderId, orderedPrice) {
        super();
        this.productId = (productId | 0);
        this.orderId = (orderId | 0);
        this.orderedPrice = orderedPrice;
    }
}

export function OrderedProductDto$reflection() {
    return record_type("MyStore.Dto.Shop.OrderedProductDto", [], OrderedProductDto, () => [["productId", int32_type], ["orderId", int32_type], ["orderedPrice", float64_type]]);
}

export class CartModel extends Record {
    constructor(cart, products, isCurrent) {
        super();
        this.cart = cart;
        this.products = products;
        this.isCurrent = isCurrent;
    }
}

export function CartModel$reflection() {
    return record_type("MyStore.Dto.Shop.CartModel", [], CartModel, () => [["cart", CartDto$reflection()], ["products", array_type(ProductDto$reflection())], ["isCurrent", bool_type]]);
}

export class OrderModel extends Record {
    constructor(order, orderedProducts) {
        super();
        this.order = order;
        this.orderedProducts = orderedProducts;
    }
}

export function OrderModel$reflection() {
    return record_type("MyStore.Dto.Shop.OrderModel", [], OrderModel, () => [["order", OrderDto$reflection()], ["orderedProducts", array_type(OrderedProductDto$reflection())]]);
}

export class ProductsModel extends Record {
    constructor(products) {
        super();
        this.products = products;
    }
}

export function ProductsModel$reflection() {
    return record_type("MyStore.Dto.Shop.ProductsModel", [], ProductsModel, () => [["products", array_type(ProductDto$reflection())]]);
}

export class CartsQuery extends Record {
    constructor(isPublic, count, offset) {
        super();
        this.isPublic = isPublic;
        this.count = (count | 0);
        this.offset = (offset | 0);
    }
}

export function CartsQuery$reflection() {
    return record_type("MyStore.Dto.Shop.CartsQuery", [], CartsQuery, () => [["isPublic", bool_type], ["count", int32_type], ["offset", int32_type]]);
}

export class CartsModel extends Record {
    constructor(carts, query) {
        super();
        this.carts = carts;
        this.query = query;
    }
}

export function CartsModel$reflection() {
    return record_type("MyStore.Dto.Shop.CartsModel", [], CartsModel, () => [["carts", array_type(CartDto$reflection())], ["query", CartsQuery$reflection()]]);
}

export class SetCurrentCartQuery extends Record {
    constructor(setCurrent) {
        super();
        this.setCurrent = setCurrent;
    }
}

export function SetCurrentCartQuery$reflection() {
    return record_type("MyStore.Dto.Shop.SetCurrentCartQuery", [], SetCurrentCartQuery, () => [["setCurrent", bool_type]]);
}

File: MyStore/MyStore.Domain/Dto/Shop.fs
namespace MyStore.Dto.Shop

open System
#if FABLE_COMPILER
open Fable.System.ComponentModel.Annotations
#else
open System.ComponentModel.DataAnnotations
#endif

type ProductDto =
    { [<Required>]
      productId: int
      [<Required>]
      name: string
      [<Required>]
      description: string
      [<Required>]
      price: double }

type CustomerDto =
    { [<Required>]
      customerId: int
      [<Required>]
      firstName: string
      lastName: string
      honorific: string
      [<Required>]
      email: string
      currentCartId: Nullable<int> }

type CartDto =
    { [<Required>]
      cartId: int
      [<Required>]
      isPublic: bool
      ownerCustomerId: Nullable<int> }

type OrderDto =
    { [<Required>]
      orderId: int
      [<Required>]
      customerId: int
      [<Required>]
      createTimeOffset: DateTimeOffset }

type OrderedProductDto =
    { [<Required>]
      productId: int
      [<Required>]
      orderId: int
      [<Required>]
      orderedPrice: double }


type CartModel =
    { [<Required>]
      cart: CartDto
      [<Required>]
      products: ProductDto array
      [<Required>]
      isCurrent: bool }

type OrderModel =
    { [<Required>]
      order: OrderDto
      [<Required>]
      orderedProducts: OrderedProductDto array }


type CartsQuery =
    { [<Required>]
      isPublic: bool
      [<Required>]
      count: int
      [<Required>]
      offset: int }

type CartsModel =
    { [<Required>]
      carts: CartDto array
      [<Required>]
      query: CartsQuery }

type SetCurrentCartQuery = { setCurrent: bool }

type ProductModel =
    { [<Required>]
      product: ProductDto
      [<Required>]
      isInCart: bool
      [<Required>]
      isLoggedIn: bool }

type ProductsQuery =
    { [<Required>]
      count: int
      [<Required>]
      offset: int }

type ProductsModel =
    { [<Required>]
      products: ProductDto array
      [<Required>]
      query: ProductsQuery }
